File: snippets/fsharp/VS_Snippets_CLR_System/system.Func~2/fs/Func2_1.fs
module Func2_1

// <Snippet2>
open System

let uppercaseString (inputString: string) =
    inputString.ToUpper()

// Instantiate delegate to reference uppercaseString function
let convertMethod = Func<string, string> uppercaseString
let name = "Dakota"

// Use delegate instance to call uppercaseString function
printfn $"{convertMethod.Invoke name}"

// This code example produces the following output:
//    DAKOTA
// </Snippet2>

File: snippets/csharp/System/FuncT,TResult/Overview/Func2_1.cs
using System;

public class GenericFunc
{
    public static void Main()
    {
        // <Snippet2>
        // Instantiate delegate to reference UppercaseString method
        Func<string, string> convertMethod = UppercaseString;
        string name = "Dakota";
        // Use delegate instance to call UppercaseString method
        Console.WriteLine(convertMethod(name));

        string UppercaseString(string inputString) => inputString.ToUpper();

        // This code example produces the following output:
        //
        //    DAKOTA
        // </Snippet2>
    }
}
